### ISSUE TITLE
Intermittent Input Loop Issue When Entering Application Name

### ISSUE THREAD

**Issue by tken2039** [argoproj/argo-cd#20392]
**Describe the bug**

I'm running into a weird issue when creating an application in Argo CD. Specifically, there’s an input loop happening when I enter the Application Name. It seems to occur more often with longer strings, but it’s not consistent. Reloading the page fixes it temporarily, and I can input the name again.

**To Reproduce**

1. Open Argo CD and start creating a new application.
2. Enter an Application Name, particularly a longer string.
3. Observe the input loop behavior.

**Expected behavior**

The application name should be entered without any looping or repetition issues.

**Version**

```shell
Argo CD version: v2.12.3
```

**Screenshots**

https://github.com/user-attachments/assets/0fda761c-e6da-446b-833a-96d84e03cd46

**Additional Context**

Has anyone else encountered this problem? Are there any known solutions or updates that address this?

Thanks!

**Comment by surajyadav1108** [comment]
Similar #19819 also, I wasn't able to reproduce the looping problem during the sync process for the application. However, the input fields do behave strangely, such as re-rendering characters that were just deleted.

**Comment by gwynforthewyn** [comment]
I'm experiencing the same issue across several instances of Argo CD. These separate instances are installed on separate clusters.

We've taken a look in the firefox browser console and don't see any network traffic being fired off.

We do have a Source Map Error:

```text
Source map error: Error: request failed with status 404
Resource URL: https://heavy-argo-cd.<sanitized the top level domain>.com/applications?showFavorites=false&proj=&sync=&autoSync=&health=&namespace=&cluster=&labels=&new=%7B%22apiVersion%22%3A%22argoproj.io%2Fv1alpha1%22%2C%22kind%22%3A%22Application%22%2C%22metadata%22%3A%7B%22name%22%3A%22%22%7D%2C%22spec%22%3A%7B%22destination%22%3A%7B%22name%22%3A%22%22%2C%22namespace%22%3A%22%22%2C%22server%22%3A%22%22%7D%2C%22source%22%3A%7B%22path%22%3A%22%22%2C%22repoURL%22%3A%22%22%2C%22targetRevision%22%3A%22HEAD%22%7D%2C%22sources%22%3A%5B%5D%2C%22project%22%3A%22%22%7D%7D
Source Map URL: ReactToastify.css.map
```

We tried upgrading from to 2.12.3 to 2.12.6 without seeing the issue go away. We see it in both versions, and across upgrades.

We did not see it in the older version we had deployed, 2.7.0.

Happy to provide any debug information you can think of that'd be useful tracking this down.

**Comment by gwynforthewyn** [comment]
We made progress over here by getting some applications that were out of sync to start synchronising, actually by removing `info` blocks (https://argo-cd.readthedocs.io/en/stable/user-guide/extra_info/) that weren't syncing properly. That allowed one of our instances' UIs to start behaving correctly.

I'm working on the other instance now: I have an externalSecret that's never synced correctly due to something to do with the decodingStrategy and conversationStrategy fields

**Comment by gwynforthewyn** [comment]
A very minor update here: after resolving every resource stuck in syncing, we _still_ have the issue going on. We do have 2 other Applications that are stuck in other polling states (Progressing...) so I'm assuming they're our issue.I'm hoping fixing those will resolve the issue.

**Comment by keithchong** [comment]
I can reproduce this problem.  This should be fixed by https://github.com/argoproj/argo-cd/pull/19717, which was cherry picked (https://github.com/argoproj/argo-cd/pull/20508) back to the 2.12 branch, and likely will be in the 2.12.7 release.

**Comment by blakepettersson** [comment]
Should be fixed with 2.12.7
